extension/sigv4authextension/extension.go

```diff
@@ -211,3 +214,13 @@ func inferRegionFromRequestURL(r *http.Request) string {
 		return ""
 	}
 }
+
+func inferServiceFromRequestURL(r *http.Request) string {
```

Comment:
With the `(.es)?` condition not added to infer region, it looks like it wouldn't be inferred for an ES request.

Let's have one 

```
func (si *SigningRoundTripper) inferServiceAndRegion(r *http.Request) (service string, region string) {
}
```

to keep the logic together.

Also, I think it makes sense to just use simpler matching of our knowns since the types of URLs that can ever be passed to the extension are well known

```
h := r.Host
if strings.HasPrefix(h, "aps-workspaces") {
  service = "aps"
  rest := h[14:]
  region = h[0, index(rest, '.')]
} else if strings.HasPrefix(h, "search-") {
  service = "es"
  ...
}
// Could fall back to a regex or just not bother
```

Reply:
Simplified the logic to use simpler matching, and also combined the functions together into one method. Unsure of what the behavior should look like if we can infer one of service/region but not the other, I'm thinking that since we're working under the assumption that the URLs that can be passed to the extension are well known, that inferring only one will never happen, so currently the logic is either neither are inferred (both returned as empty string) or both are inferred